### PR TITLE
Doxygen: stop maintaining old versions, align around new version

### PR DESCRIPTION
**Pull request by jcar87**
### Summary
Changes to recipe:  doxygen/all

#### Details
* Stop publishing revisions for older versions
* Use version range elswhere when requiring doxygen
* Use `cmake_additional_variables_prefixes` to handle iconv instead of patching
* Add version 1.12.0

### CODE REVIEW

File: recipes/doxygen/all/conanfile.py [hunk 1]
```diff
@@ -56,28 +41,34 @@ def requirements(self):
...
 
     def build_requirements(self):
-        if self._settings_build.os == "Windows":
+        if self.settings_build.os == "Windows":
             self.tool_requires("winflexbison/2.5.24")
         else:
             self.tool_requires("flex/2.6.4")
             self.tool_requires("bison/3.8.2")
 
+        self.tool_requires("cmake/[>=3.19 <4]")
```
Comment (jcar87) [inline recipes/doxygen/all/conanfile.py:62]: This is to fix an issue seen on Windows  - newer versions of CMake have a more correct `find_package(Python)`

**Comment by valgur** [comment]
You should probably skip the ignition-* libraries for now since they have not yet been migrated to Conan 2.x.

**Review by franramirez688 — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM. Doxygen was a pain in COnan 1.x due compatibility with glibc version used to build and cppstd. With this change, only one compiler is being used (GCC11) and the cppstd is followed what is used by Conan. 
Nice job.

File: recipes/doxygen/all/conanfile.py [hunk 2]
```diff
@@ -56,28 +41,34 @@ def requirements(self):
...
+
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        apply_conandata_patches(self)
+
+        #Do not build manpages
+        cmakelists = os.path.join(self.source_folder, "CMakeLists.txt")
+        replace_in_file(self, cmakelists, "add_subdirectory(doc)", "")
+        replace_in_file(self, cmakelists, "set(CMAKE_CXX_STANDARD", "##set(CMAKE_CXX_STANDARD")
```
[the remark this answers is not in the document]
Comment (uilianries) [reply recipes/doxygen/all/conanfile.py:71]: Good catch!

**Comment by uilianries** [comment]
This PR probably fixes a bunch of known issues in CCI!

**Review by AbrilRBS — APPROVED** [review]

**Comment by jcar87** [comment]
> LGTM. Doxygen was a pain in COnan 1.x due compatibility with glibc version used to build and cppstd. With this change, only one compiler is being used (GCC11) and the cppstd is followed what is used by Conan. Nice job.

I don't think this PR moves the needle on glibc or gcc version  - the package_id will take into account the full compiler settings _and_ the dependencies - so anybody using a different gcc version will end up with missing binaries, and the glibc version is the same as in the base docker images. 
If we want a single binary package regardless of compiler setting and dependencies versions (since we can probably assume all are equivalent thus there's little need for variation) we could probably do:
- `vendor=True`
- `*:shared!=False` ("important" options)

Which I think would cause the dependency graph for this executable to be detached from anything else. Doesn't solve the glibc problem for older systems, but at the same time 8 years is already reasonably old :D 

> You should probably skip the ignition-* libraries for now since they have not yet been migrated to Conan 2.x.

Thanks! Probably not a bad thing to merge as is since it will ensure any PR attempting that will take into account those changes. No recipe revision will be published since the recipes are not compatible with Conan 2 yet.

**Pull request merged**